packages/smart-contracts/test/contracts/EthereumProxy.test.ts

```diff
@@ -37,20 +39,18 @@ describe('contract: EthereumProxy', () => {
     const fromOldBalance = await provider.getBalance(from);
     const toOldBalance = await provider.getBalance(to);
 
-    const receipt = await (
+    await (
       await ethProxy.transferWithReference(to, referenceExample, {
         value: amount,
         gasPrice: DEFAULT_GAS_PRICE,
       })
     ).wait();
 
-    const gasCost = DEFAULT_GAS_PRICE.mul(BigNumber.from(receipt.gasUsed));
-
     const fromNewBalance = await provider.getBalance(from);
     const toNewBalance = await provider.getBalance(to);
 
     // Check balance changes
-    expect(fromNewBalance.toString()).to.equals(fromOldBalance.sub(gasCost).sub(amount).toString());
+    expect(fromNewBalance).to.be.lt(fromOldBalance.sub(amount));
```

Comment:
If we cannot test the value precisely, we should at least add a lower boundery such as `to.be.gt(fromOldBalance.sub(amount).mul(99).div(100))`